include/iterlib/FilterIterator.h

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include "iterlib/WrappedIterator.h"
+
+namespace iterlib {
+
+enum class FilterType {
+  INVALID = 0,
+  GE,
+  GT,
+  LE,
+  LT,
+  RANGE,
+  EQ,
+  NE,
+  EXISTS,
+  PREFIX,
+  INSET,
+  CONTAINS,
+};
+
+class FilterIteratorBase : public WrappedIterator {
+ public:
+  explicit FilterIteratorBase(Iterator* iter) : WrappedIterator(iter) {}
+
+  bool orderPreserving() const override { return true; }
+
+ protected:
+  bool doNext() override;
+
+  bool doSkipTo(id_t id) override;
+
+  virtual bool match(const Iterator* iter) = 0;
+};
+
+class FilterIterator : public FilterIteratorBase {
+ public:
+  explicit FilterIterator(Iterator* iter)
+      : FilterIteratorBase(iter), firstTime_(true) {}
+
+  void setFilter(const std::vector<std::string>& fields,
+                 const std::vector<dynamic>& values, FilterType filterType) {
+    if (fields.size() == 1) {
+      fields_ = fields[0];
+    } else {
+      auto fieldVec = variant::vector_dynamic_t();
+      for (const auto& v : fields) {
+        fieldVec.emplace_back(v);
+      }
+      fields_ = std::move(fieldVec);
+    }
+
+    // We support operators with:
```

Comment:
This piece of the comment is quite confusing because there is no s expression language that we yet have to describe the filter iterator.